app/models/oil_type.rb

```diff
@@ -1,3 +1,12 @@
 class OilType < ActiveRecord::Base
   has_many :chips
+  validates :name, presence: true, uniqueness: true
+
+  def name_id
+    name.downcase.gsub(" ", "_")
+  end
+
+  def self.storage_name(downcased_name)
+    downcased_name.gsub("_", " ").split.map(&:capitalize).join(' ')
```

Comment:
Use `tr` instead of `gsub`.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.